lib/mirage.ml

```diff
@@ -1739,14 +1739,16 @@ let compile libs warn_error target =
   Bos.OS.Cmd.run cmd
 
 (* Implement something similar to the @name/file extended names of findlib. *)
-let rec expand_name ~lib param =
+let rec expand_name param =
+  let pkg_path name = query_ocamlfind ~format:"%d" [name] >>| List.hd in
```

Comment:
hmm, this `List.hd` can fail, or!?  since this is already in a `result` type, shouldn''t it rather be:
```
query_ocamlfind ~format:"%d" [name] >>= function
| x::_ -> R.ok x
| _ -> Error (`Msg "???")
```

but what is the actual semantics?  is `ocamlfind` in this case expected to return exactly one item?